pkg/model/project.go

```diff
@@ -243,3 +322,73 @@ func (p *ProjectSSOConfig_GitHub) GenerateAuthCodeURL(project, callbackURL, stat
 
 	return authURL, nil
 }
+
+// UpdateRBACRoles update only custom rbac roles.
+func (p *Project) UpdateRBACRoles(roles []*ProjectRBACRole) error {
+	v := make(map[string]struct{}, len(roles))
+	update := make([]*ProjectRBACRole, 0, len(roles))
+	for _, role := range roles {
+		if role.IsBuiltin {
+			continue
+		}
+		if role.IsBuiltinName() {
+			return fmt.Errorf("cannot use built-in role name")
+		}
+		if _, ok := v[role.Name]; ok {
+			return fmt.Errorf("role name must be unique")
+		}
+		v[role.Name] = struct{}{}
+		update = append(update, role)
+	}
+	p.RbacRoles = update
+	return nil
+}
+
+func (p *ProjectRBACRole) IsBuiltinName() bool {
+	return p.Name == builtinRBACRoleAdmin.String() ||
+		p.Name == builtinRBACRoleEditor.String() ||
+		p.Name == builtinRBACRoleViewer.String()
+}
+
+// UpdateUserGroups update user groups.
+func (p *Project) UpdateUserGroups(groups []*ProjectUserGroup) error {
+	v := make(map[string]struct{}, len(groups))
+	for _, group := range groups {
+		if _, ok := v[group.SsoGroup]; ok {
+			return fmt.Errorf("cannot assign multi roles to sso group")
+		}
+		v[group.SsoGroup] = struct{}{}
+	}
+	p.UserGroups = groups
+	return nil
+}
+
+// SetUserGroup add user group or update user group if already exists.
+func (p *Project) SetUserGroup(group, role string) {
+	v := &ProjectUserGroup{
+		SsoGroup: group,
+		Role:     role,
+	}
+	for i := range p.UserGroups {
+		if p.UserGroups[i].SsoGroup == group {
+			p.UserGroups[i] = v
+			return
+		}
+	}
+	p.UserGroups = append(p.UserGroups, v)
+}
+
+// MigrateFromRBAC migrate rbac config to user group.
+func (p *Project) MigrateFromRBAC() {
+	rbac := p.Rbac
+	if rbac.GetAdmin() != "" {
+		p.SetUserGroup(rbac.Admin, builtinRBACRoleAdmin.String())
+	}
+	if rbac.GetEditor() != "" {
+		p.SetUserGroup(rbac.Editor, builtinRBACRoleEditor.String())
+	}
+	if rbac.GetViewer() != "" {
+		p.SetUserGroup(rbac.Viewer, builtinRBACRoleViewer.String())
+	}
+	p.Rbac = &ProjectRBACConfig{}
+}
```

Comment:
Sorry, why do we need this migrate function?
I think at the last MTG, we finally decided to build the list of UserGroups based on the both new RBAC and old RBAC instead of migrating between them. 🤔

Reply:
I was gonna migrate new RBAC from old one when it comes to mapping. (excepting RBAC role.)
Because I feel that it sometime cause complexity by existing both of `ProjectRBACRole` and `ProjectUserGroup`.
For instance, we can assign built-in role into multi user groups and then it's hard to identify whether we have to update  `ProjectRBACRole` or`ProjectUserGroup`. but wdyt?

Reply:
May I ask where this function will be called?
I feel that by adding this migrate function, the hard part will be deciding when to call it.
We have to ensure that it must be called exactly once before using, right?

Reply:
I guess that this function should be called through [this](https://github.com/pipe-cd/pipecd/blob/master/pkg/datastore/projectstore.go#L213) when ops server start.
And I think it's okay because this method guarantee idempotency.

Reply:
@knanao  I got your point. But this will override the user-specified group with the same group name as the current implementation of the `SetUserGroup` function.
Is the `SetUserGroup` function only be used inside `MigrateFromRBAC`, or other places too?

Reply:
@nghialv 
Currently, it's supposed to be only used inside `MigrateFromRBAC`.
I made it the method of `Project` for readability. 
But Is it better that that implementation should be confined in `MigrateFromRBAC` or add comment which indicate only used from that?.

Reply:
Finally, I removed migration method to be more simple.
Merge old RBAC and new RBAC when returning a response and update them and remove old one when updating.
Thank you for your reviewing.